### ISSUE TITLE
Drag & dropping items with the mouse besides a dialog should drop them to the ground

### ISSUE THREAD

**Issue by flo** [MovingBlocks/Terasology#2218]
Drag & dropping items with the mouse besides a dialog should drop them to the ground.

Currently the only way to clear an inventory is via hotkey which is not user friendly.

**Comment by flo** [comment]
Ideal would be a solution that works for all modules.

If none can be found, it would be acceptable for now just to have the inventory dialog actually convering the whole screen with an invisible background area that accepts drops from the mouse and makes the item drop to the ground.

**Comment by genuinelucifer** [comment]
@flo I was just looking for a place to start working on the issue and found that ```InventoryScreen.java``` already exists. What I couldn't find was how to activate/open it once in the game. Sorry, if it's silly. I opened the ```Builder Sample Gameplay``` and with right click I could drop only cubes and with left click erase them. What are the hotkeys for other shapes?
Is there some fundamental wiki page where I could learn about building gameplays?

**Comment by genuinelucifer** [comment]
Figuring out this one is seeming to be more difficult than I accepted.
.
Firstly, when I did ```giveItem Core:TNT 5``` I recieved 5 TNT blocks which I could see in the slots when I pressed ```i``` to open the inventory. Now, I can only see sand in those boxes If I collect sand. If I give myself TNT or anything else it is NOT being shown in the inventory, rather If I scroll or use the numbers 0-9 then I am able to use TNT but can't see it in the inventory!!
What am I missing?
.
Also, I was looking for the codes where mouse drag drops of slot Items are handled. I went through ```InventoryCell.java``` which apparently has an ```InteractionListener``` but the drag drop is NOT handled there. 
Then I went ahead to check if ```InventoryUtils`` is doing this. I placed ```System.out.println``` in almost all the functions to figure out the order in which the functions are called once I dragged and dropped the Sand in inventory screen. But, NO output could be seen from the println statements!!!
Then I went ahead and did the same thing with some other classes related to inventory but couldn't figure it out.
.
Pleeease help..... atleast to find out where the drag drop on inventory screen is handled. And why can't I see anything else except sand/dust on my inventory screen?!

**Comment by flo** [comment]
Builder Sample GAmplay is the wrong place to start as it is an example for a mode without inventory...

There is a hotkey for opening the inventory. The hotkey is I.

**Comment by genuinelucifer** [comment]
@flo Sorry for not mentioning this. I got the solution to that via the irc channel. Now, I am able to open inventory but not always do I see the items in inventory grid. Also some of other problems that i mentioned.

**Comment by flo** [comment]
I suggest you view the call hierachy (IDE function) of the swapItem, giveAmount  and takeAmount methods of InventoryCell

**Comment by genuinelucifer** [comment]
Thanks. Will try. :)

**Comment by flo** [comment]
oh and interactionListener of InventorCell does handle the drop. I set a breakpoint to verify it

**Comment by genuinelucifer** [comment]
Thanks.

**Comment by genuinelucifer** [comment]
I finally found out that NOT every file is re-compiled when I run the program.
So, I did  a full rebuild and stumbled upon looots of errors. All of them in ```engine-tests```
.
```org.junit.*``` could not be resolved. So, I downloaded the jar and put it into libs folder for the project. 
```org.jboss.shrinkwrap``` could not be resolved. So, I did the same for this.
Then I did sync the project.
.
But, the errors still persist!! How do I resolve these errors?
Please help.

**Comment by msteiger** [comment]
Which IDE are you using? You might want to recreate your project files, e.g. `gradlew idea` or `gradlew eclipse`. Alternatively you can add JUnit and the other libs. manually to see if that resolves your problem.

**Comment by genuinelucifer** [comment]
I am using intellij idea.
I did do the gradlew idea command. But still the error persists. And adding
the jar files manually too didn't help.

**Comment by genuinelucifer** [comment]
Just an update. I did the build again (after putting the jars) using ```gradlew game``` and then ```gradlew idea``` and it built successfully.
Then once I opened intellij, i built it first (successfully). Then removed the jars (deleted them) and then built again and it built perfectly!!
I don't know what happened there.  But finally it is working and hopefully I'll submit a PR for this issue within 1-2 days... :)
